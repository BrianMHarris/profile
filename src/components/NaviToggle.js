import React from 'react';
import './NaviToggle.css';

// The toggle button for collapsing menu items on smaller screens
// Props:
//  handleClick - the function passed from the parent down to the button
//  logo - the svg item to pass down to the button
//  label - the text displayed on the button, only appears if no imgURL provided
//  isOpen - whether the menu is open or not

const NaviToggle = (props) => {
  // a silly wrapper to make sure preventDefault is called
  function handleClick(e) {
    e.preventDefault();
    props.handleClick();
  }

  function displayToggle(isOpen) {
    if (isOpen) {
      return (
        <img src={props.openLogo} alt={props.label}/>
      )
    }
    return (
      <img src={props.logo} alt={props.label}/>
    )
  }

  return (
    <div className="navitoggle-std"
          onClick={handleClick}>
      {displayToggle(props.isOpen)}
    </div>
  )
};

export default NaviToggle;
