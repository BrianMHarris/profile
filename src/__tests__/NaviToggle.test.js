import React from 'react';
import ReactDOM from 'react-dom';
import NaviToggle from '../components/NaviToggle';
import renderer from "react-test-renderer";

describe('NaviToggle', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NaviToggle />, div);
  });

  it('matches the snapshot', () => {
    const component = renderer.create(
      <NaviToggle />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
