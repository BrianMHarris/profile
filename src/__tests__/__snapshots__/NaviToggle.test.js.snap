// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NaviToggle matches the snapshot 1`] = `
<div
  className="navitoggle-std"
  onClick={[Function]}
>
  <img
    alt={undefined}
    src={undefined}
  />
</div>
`;
